packages/hash/frontend/src/components/layout/PageHeader/SearchBar/SearchInput.tsx

```diff
@@ -0,0 +1,204 @@
+        <IconButton
+          sx={(theme) => ({
+            width: theme.spacing(2.5),
+            height: theme.spacing(2.5),
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "center",
+            // @todo figure out how to fetch this from `theme`
+            borderRadius: "2px",
+            color: theme.palette.gray[50],
+
+            "&:hover": {
+              transition: theme.transitions.create([
+                "color",
+                "background-color",
+              ]),
+              backgroundColor: theme.palette.gray[20],
+              color: theme.palette.gray[80],
+            },
+          })}
+          onClick={clearSearch}
+        >
+          <FontAwesomeSvgIcon icon={faXmark} />
+        </IconButton>
+      </Tooltip>
+    </Box>
+  );
+};
+
+const ShortcutIcon = () => {
+  const isMac = navigator.userAgent.toUpperCase().includes("MAC");
+
+  return (
+    <Box
+      sx={(theme) => ({
+        marginRight: theme.spacing(1),
+        height: "100%",
+        pointerEvents: "none",
+        display: "flex",
+        alignItems: "center",
+        fontWeight: "bold",
+      })}
+    >
+      <Box
+        sx={(theme) => ({
+          height: "26px",
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "center",
+          color: theme.palette.gray[50],
+          backgroundColor: theme.palette.gray[20],
+          borderRadius: "2px",
+          px: 1,
+          py: 0.75,
+        })}
+      >
+        <Typography variant="microText">
+          {isMac ? "Cmd" : "Ctrl"} + P
+        </Typography>
+      </Box>
+    </Box>
+  );
+};
+
+export const SearchInput: React.FC<{
+  displayedQuery: string;
+  isMobile: boolean;
+  setResultListVisible: (visible: boolean) => void;
+  setQueryText: (queryText: string) => void;
+}> = ({ displayedQuery, isMobile, setResultListVisible, setQueryText }) => {
+  const isMac = navigator.userAgent.toUpperCase().includes("MAC");
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useKeys(["ControlLeft", "KeyP"], (event) => {
+    event.preventDefault();
+    if (!isMac) {
+      inputRef.current?.focus();
+    }
+  });
+
+  useEffect(() => {
+    function checkSearchKey(event: KeyboardEvent) {
+      if (isMac && event.key === "p" && event.metaKey) {
+        event.preventDefault();
+        inputRef.current?.focus();
+      }
+    }
+
+    document.addEventListener("keydown", checkSearchKey);
+
+    return () => {
+      document.removeEventListener("keydown", checkSearchKey);
+    };
+  }, [isMac]);
+
+  const clearSearch = useCallback(() => {
+    setQueryText("");
+  }, [setQueryText]);
+
+  return (
+    <Box
+      sx={(theme) => ({
+        position: "relative",
+        borderRadius: theme.spacing(0.75),
+        borderWidth: 1,
+        borderStyle: "solid",
+        borderColor: theme.palette.gray[30],
+        width: isMobile ? "100%" : "385px",
```

Comment:
Making the input fixed-width on desktop to better support variable-width `endAdornment`s.

Reply:
This is still needed after the `OutlinedInput` refactor, as the width of the 2 possible adornments, ie, shortcut key or cancel icon, is different. The fixed width helps in preventing a layout shift when switching between the two.